extensions/integration-tests/src/notebook.test.ts

```diff
@@ -14,136 +14,184 @@ import { sqlNotebookContent, writeNotebookToFile, sqlKernelMetadata, getFileName
 import { getBdcServer, getConfigValue, EnvironmentVariable_PYTHON_PATH } from './testConfig';
 import { connectToServer } from './utils';
 import * as fs from 'fs';
+import { stressify, sleep } from '../testfmks/src/stress';
 
 if (context.RunTest) {
 	suite('Notebook integration test suite', function () {
-		setup(function () {
+		setup(async function () {
+			sleep(20000);
+			console.log(`environment variable SuiteType is set to ${process.env.SuiteType}`);
 			console.log(`Start "${this.currentTest.title}"`);
+			let server = await getBdcServer();
+			assert(server && server.serverName, 'No server could be found');
+			await connectToServer(server, 6000);
 		});
 		teardown(async function () {
-			let testName = this.currentTest.title;
-			try {
-				let fileName = getFileName(testName);
-				if (fs.existsSync(fileName)) {
-					fs.unlinkSync(fileName);
-					console.log(`"${fileName}" is deleted.`);
-				}
-				await vscode.commands.executeCommand('workbench.action.closeActiveEditor');
-			}
-			catch (err) {
-				console.log(err);
-			}
-			finally {
-				console.log(`"${testName}" is done`);
-			}
+			await (new NotebookTester()).cleanup(this.currentTest.title);
```

Comment:
Going back to what I said in the previous test; Can we not have a separate class just for tests? It would suck to have to do this unnecessary code move just to be able to stress a test.

Reply:
No we cannot - unfortunately we do need a separate class for test. It is a requirement imposed by Decorators.

Reply:
We will discuss this on the design review.

Reply:
Couldn't you just wrap the method body in an lambda and decorate that instead of having to create a whole separate class for it?

Reply:
decorators work only on class methods and do not work on plain functions or anonymous functions

Reply:
As Kevin was saying yesterday though it would be awesome if instead of splitting the test functions and test implementation into two separate parts we just combined it so that the entire test was contained in the class at the very least then. I just want to try and avoid adding more code cluttering up these tests than is necessary.

Reply:
Yes it would be awesome but we can come back and do that later. This is not a current priority.

---
In reply to: [286147185](https://github.com/microsoft/azuredatastudio/pull/5499#discussion_r286147185) [](ancestors = 286147185)